plugins/inputs/nats_consumer/nats_consumer.go

```diff
@@ -0,0 +1,202 @@
+package natsconsumer
+
+import (
+	"fmt"
+	"log"
+	"sync"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/plugins/parsers"
+	"github.com/nats-io/nats"
+)
+
+type natsError struct {
+	conn *nats.Conn
+	sub  *nats.Subscription
+	err  error
+}
+
+func (e natsError) Error() string {
+	return fmt.Sprintf("%s url:%s id:%s sub:%s queue:%s",
+		e.err.Error(), e.conn.ConnectedUrl(), e.conn.ConnectedServerId(), e.sub.Subject, e.sub.Queue)
+}
+
+type natsConsumer struct {
+	QueueGroup string
+	Subjects   []string
+	Servers    []string
+	Secure     bool
+
+	PointBuffer int
+	parser      parsers.Parser
+
+	sync.Mutex
+	Conn *nats.Conn
+	Subs []*nats.Subscription
+
+	// channel for all incoming NATS messages
+	in chan *nats.Msg
+	// channel for all NATS read errors
+	errs chan error
+	// channel for all incoming parsed points
+	metricC chan telegraf.Metric
+	done    chan struct{}
+}
+
+var sampleConfig = `
+  ### urls of NATS servers
+  servers = ["nats://localhost:4222"]
+  ### Use Transport Layer Security
+  secure = false
+  ### subject(s) to consume
+  subjects = ["telegraf"]
+  ### name a queue group
+  queue_group = "telegraf_consumers"
+  ### Maximum number of points to buffer between collection intervals
+  point_buffer = 100000
```

Comment:
can you change `point_buffer` to `metric_buffer`? I have a PR making this change in the kafka consumer already